shell/platform/common/cpp/client_wrapper/include/flutter/event_stream_handler.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_EVENT_STREAM_HANDLER_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_EVENT_STREAM_HANDLER_H_
+
+#include "event_sink.h"
+
+namespace flutter {
+
+// Handler of stream setup and tear-down requests.
+// Implementations must be prepared to accept sequences of alternating calls to
+// onListen() and onCancel(). Implementations should ideally consume no
+// resources when the last such call is not onListen(). In typical situations,
+// this means that the implementation should register itself with
+// platform-specific event sources onListen() and deregister again onCancel().
+template <typename T>
+struct StreamHandler {
```

Comment:
What is the argument for this being a struct of function objects rather than—as on the other platforms—an interface? Using different designs for the different language variants adds some conceptual overhead for people dealing with multiple variants, so I'd like to be sure there's a compelling reason for this difference.

Reply:
@stuartmorgan 

At the point of [the review](https://github.com/flutter/engine/pull/17015#discussion_r391850887) , I changed from class to struct. 

In addition, MethodChannel is implemented using function, so I thought that it would be better to use function in the same way. I could only think of this to use function with one argument. Is it better to use Interface Class? 

I hope this answers your question.

[method_channel.h](https://github.com/flutter/engine/blob/master/shell/platform/common/cpp/client_wrapper/include/flutter/method_channel.h)
```
using MethodCallHandler =
    std::function<void(const MethodCall<T>& call,
                       std::unique_ptr<MethodResult<T>> result)>;

void SetMethodCallHandler(MethodCallHandler<T> handler) const {
```

Reply:
Since the last review, I needed to address the same thing for MethodResult. For consistency, please follow the pattern I used there of having [an interface class](https://github.com/flutter/engine/blob/9992a4ae664b0de4bf1834ad44dc48c7ee6025e8/shell/platform/common/cpp/client_wrapper/include/flutter/method_result.h), and then a [function-based implementation](https://github.com/flutter/engine/blob/9992a4ae664b0de4bf1834ad44dc48c7ee6025e8/shell/platform/common/cpp/client_wrapper/include/flutter/method_result_functions.h) for cases where implementing the interface isn't convenient.

Reply:
OK, thank you.